packages/core/src/components/widgets/FileWidget.tsx

```diff
@@ -150,20 +147,14 @@ function FileWidget<
 
   return (
     <div>
-      <p>
-        <input
-          id={id}
-          name={id}
-          type="file"
-          disabled={readonly || disabled}
-          onChange={handleChange}
-          defaultValue=""
-          autoFocus={autofocus}
-          multiple={multiple}
-          accept={options.accept ? String(options.accept) : undefined}
-          aria-describedby={ariaDescribedByIds<T>(id)}
-        />
-      </p>
+      <BaseInputTemplate
+        {...props}
+        disabled={disabled || readonly}
+        type="file"
+        onChangeOverride={handleChange}
```

Comment:
How were you able to supply this `onChangeOverride` prop without adding it to the `WidgetProps` type?

Reply:
Is file upload the only case that would benefit from this style?

Having both `onChange` and `onChangeOverride` seems like a confusing API, but I'm not sure if it should be clarified with better prop names or if it should have a different design.